src/custom/components/Web3ReactManager/index.tsx

```diff
@@ -48,6 +55,12 @@ export default function Web3ReactManager({ children }: { children: JSX.Element }
     }
   }, [])
 
+  useEffect(() => {
+    if (referralLink) {
+      dispatch(updateAffiliateLink({ affiliateLink: referralLink }))
+    }
+  }, [account, referralLink, dispatch])
+
```

Comment:
We shouldn't be updating either moving this Web3ReactManager file. Create a new hook which does this integration and put it in the App/index.ts file so it runs when the app renders.